truffle/src/main/java/org/jruby/truffle/nodes/core/TrufflePrimitiveNodes.java

```diff
@@ -541,6 +541,28 @@ public Object synchronize(VirtualFrame frame, DynamicObject self, DynamicObject
         }
     }
 
+    @CoreMethod(names = "print_backtrace", onSingleton = true)
+    public abstract static class PrintBacktraceNode extends CoreMethodNode {
+
+        public PrintBacktraceNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        @TruffleBoundary
+        @Specialization
+        public DynamicObject printBacktrace() {
+            final List<String> rubyBacktrace = BacktraceFormatter.createDefaultFormatter(getContext())
+                    .formatBacktrace(null, RubyCallStack.getBacktrace(this));
+
+            for (String line : rubyBacktrace) {
+                System.err.println(line);
+            }
```

Comment:
It's a debug method - if you are debugging you may not want to start running arbitrary, complex code like `puts`, so this does it for you. We do have `debug_print` though.